### ISSUE TITLE
Document branch naming convention / other dev workflow

### ISSUE THREAD

**Issue by nikitakit** [pioneers/PieCentral#5]
I think PiE should take advantage of GitHub's new protected branch feature to avoid requiring every staff member to create their own fork. Instead, all branches can be hosted in the central repo (with master suitably protected against un-approved changes)

However, it would be good to have a documented branch convention in place to avoid an explosion of randomly-named branches. I propose something like `name/feature_name`, though there can be variation to split by project too. (Slashes are the preferred separator for branch names)

**Comment by onibrow** [comment]
I second this. Travis-ci naturally flows well with protected branches as well.

**Comment by vdonato** [comment]
Yep. I think having this documentation (probably along with build system docs) in the top level README of this repo would make the most sense.
